test/ShapeCasts.test.js

```diff
@@ -278,6 +278,58 @@ function runSuiteWithOptions( defaultOptions ) {
 
 	} );
 
+	describe( 'Bvhcast', () => {
+
+		let bvhA = null;
+		let bvhB = null;
+		let matrix;
+
+		beforeAll( () => {
+
+			const cubeA = new BoxBufferGeometry( 2, 2, 2 );
+			bvhA = new MeshBVH( cubeA );
+			const cubeB = new BoxBufferGeometry( 2, 2, 2 );
+			bvhB = new MeshBVH( cubeB );
+			matrix = new Matrix4();
+
+		} );
+
+		it( 'should test all combination of geometries triangles', () => {
+
+			matrix.makeTranslation( 1, 1, 1 );
+			let nbTriangleTests = 0;
+			const intersectsTriangles = function () {
+
+				nbTriangleTests += 1;
+				return false;
+
+			};
+
+			bvhA.bvhcast( bvhB, matrix, { intersectsTriangles: intersectsTriangles } );
+
+			expect( nbTriangleTests ).toBe( 144 );
```

Comment:
I suppose it depends on the complexity of the geometry and BVH but in practice this _wouldn't_ be testing all combinations of triangles because you want to limit the number of checks as much as possible which the BVH should help with. The number of triangles in the cube (12) is probably not enough to create a useful BVH, though. Maybe using two reasonably complex spheres and ensuring that less than the max number of triangle tests are performed would be better?

Reply:
Yes, of course. But in this example it should takes into account all triangles, my point was to create a least a dummy test that should be true.
I can still add another test (or replace if you want to) with more complex geometries, but sure, difficult to find the exact number of triangles intersected on papers.

Reply:
I see -- lets add another one, then, that checks to make sure only a subset of all triangle checks are run which is really the intent of the function.